eradiate/scenes/measure.py

```diff
@@ -479,33 +475,30 @@ def kernel_dict(self, ref=True):
         from eradiate.kernel.core import ScalarTransform4f
 
         target = self.target.to(kdu.get("length")).magnitude
-        distance = self.distance.to(kdu.get("length")).magnitude
-        zenith = self.zenith.to("rad").magnitude
-        azimuth = self.azimuth.to("rad").magnitude
-
-        origin = spherical_to_cartesian(distance, zenith, azimuth)
-        direction = origin / np.linalg.norm(origin)
+        origin = self.origin.to(kdu.get("length")).magnitude
 
         if np.allclose(origin, target):
             raise ValueError("target is too close to the camera")
 
-        up = [np.cos(azimuth), np.sin(azimuth), 0] \
-            if np.allclose(direction, [0, 0, 1]) \
+        direction = origin - target
+
+        self.up = [1, 0, 0] \
+            if np.dot(direction, [0, 0, 1]) == 0\
             else [0, 0, 1]
```

Comment:
You're ignoring the value passed by the user for `self.up` here contrary to what you wrote in the docstring. I think the simplest is to drop that part and let the user set a reasonable value for `up` themselves.

Or, if you want to keep this behaviour, you can have it handled by a special value (_e.g._ `None`). In that case, it's probably best to implement it as part of the post-init step (`kernel_dict()` should preferably not alter the object's state). As you prefer (you'd have to document that ofc).